device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/NoRetry.java

```diff
@@ -8,12 +8,17 @@
 package com.microsoft.azure.sdk.iot.device.transport;
 
 import com.microsoft.azure.sdk.iot.device.exceptions.TransportException;
+import lombok.extern.slf4j.Slf4j;
 
 /**
   * Represents a retry policy that performs no retries.
   */
+@Slf4j
 public class NoRetry implements RetryPolicy
 {
+    public NoRetry() {
+        log.info("NOTE: A new instance of {} has been created, the client will not perform any retries on disconnect.", NoRetry.class.getName());
```

Comment:
Since the name of these classes shouldn't change, you should hard code the name rather than get it through reflection each time